aya/src/maps/perf/async_perf_event_array.rs

```diff
@@ -98,16 +98,17 @@ impl<T: BorrowMut<MapData> + Borrow<MapData>> AsyncPerfEventArray<T> {
         index: u32,
         page_count: Option<usize>,
     ) -> Result<AsyncPerfEventArrayBuffer<T>, PerfBufferError> {
-        let buf = self.perf_map.open(index, page_count)?;
+        let Self { perf_map } = self;
```

Comment:
you love a destructuring don't you 😅 In this case, how is it better than just doing self.perf_map? It's the only field used and it's used only once

Reply:
I sure do! It's just an assertion that no fields have been added and don't need to be dealt with. It's a bit of paranoia that protects this code from future changes that affect semantics.